src/common/AndroidUtils.ts

```diff
@@ -196,6 +196,25 @@ export class AndroidSDKUtils {
         });
     }
 
+    public static fetchEmulators(): Promise<AndroidVirtualDevice[]> {
+        return new Promise((resolve, reject) => {
+            let devices: AndroidVirtualDevice[] = [];
+            try {
+                const result = execSync(
+                    AndroidSDKUtils.AVDMANAGER_COMMAND + ' list avd'
+                );
+                if (result) {
+                    devices = AndroidVirtualDevice.parseRawString(
+                        result.toString()
+                    );
+                }
+            } catch (exception) {
+                AndroidSDKUtils.logger.warn(exception);
```

Comment:
@maliroteh-sf Sorry for the jolt in time travel. Sometimes I find it helpful to ask questions in the context of where they ultimately originated. Do you happen to remember if we originally had a good reason to soft-play exceptions in the `device:list` command? I was running it recently to verify some changes, and realized that when I didn't have my `ANDROID_SDK_ROOT` variable set, the command just quietly continued and stated that I had no devices.

This original code may have been back when we actually ran `setup` preceding each of these actions—where those kinds of "pre-req" errors would have overwhelmingly been pre-filtered—so maybe it slipped through the cracks when we removed the checks? Anyway, it presented odd behavior in the present day, and I figured I'd ask while I was in there.

Reply:
So when I was adding this command, I did present to the team and had a team discussion around this, and the decision was to not have this command do any `setup` checks.

The reason is that this command is used in VSCode extension. When a user wants to preview a component locally and they choose to preview on an iOS or Android device, we then use this command to get a list of devices and present them in a list to the user.

Initially I had the device list command do `setup` checks and it was taking a long time (~30 secs) for it to produce results. After team discussion we decided to not run any checks and have the command return an empty list if it hits any errors. This way the user experience would be better and the user wouldn't have to wait a long time until they're presented with the list. The team decided to do this b/c we have documentation for getting started with our SFDX extensions and in that doc the first step is running the `setup` checks so we felt that by the time they make it to `device list` command most of the time they have already gotten past the `setup` phase.